problem-spring-webflux/src/main/java/org/zalando/problem/spring/webflux/advice/AdviceTrait.java

```diff
@@ -0,0 +1,157 @@
+package org.zalando.problem.spring.webflux.advice;
+
+import org.apiguardian.api.API;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.reactive.accept.HeaderContentTypeResolver;
+import org.springframework.web.server.ServerWebExchange;
+import org.zalando.problem.Problem;
+import org.zalando.problem.Status;
+import org.zalando.problem.StatusType;
+import org.zalando.problem.ThrowableProblem;
+import org.zalando.problem.spring.common.AdviceTraits;
+import org.zalando.problem.spring.webflux.advice.custom.CustomAdviceTrait;
+import org.zalando.problem.spring.webflux.advice.general.GeneralAdviceTrait;
+import org.zalando.problem.spring.webflux.advice.http.HttpAdviceTrait;
+import org.zalando.problem.spring.webflux.advice.validation.ValidationAdviceTrait;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+import static javax.servlet.RequestDispatcher.ERROR_EXCEPTION;
+import static org.apiguardian.api.API.Status.STABLE;
+
+/**
+ * <p>
+ * Advice traits are simple interfaces that provide a single method with a default
+ * implementation. They are used to provide {@link ExceptionHandler} implementations to be used in
+ * {@link Controller Controllers} and/or in a {@link ControllerAdvice}. Clients can choose which traits they what to
+ * use à la carte.
+ * </p>
+ * <p>
+ * Advice traits are grouped in packages, based on they use cases. Every package has a composite advice trait that
+ * bundles all traits of that package. Additionally there is one {@link ProblemHandling major composite advice trait}
+ * that in turn bundles all other composites.
+ * </p>
+ *
+ * @see ControllerAdvice
+ * @see ExceptionHandler
+ * @see Throwable
+ * @see Exception
+ * @see Problem
+ * @see ProblemHandling
+ * @see CustomAdviceTrait
+ * @see GeneralAdviceTrait
+ * @see HttpAdviceTrait
+ * @see ValidationAdviceTrait
+ */
+@API(status = STABLE)
+public interface AdviceTrait extends org.zalando.problem.spring.common.AdviceTrait {
+
+    /**
+     * Creates a {@link Problem problem} {@link ResponseEntity response} for the given {@link Throwable throwable}
+     * by taking any {@link ResponseStatus} annotation on the exception type or one of the causes into account.
+     *
+     * @param throwable exception being caught
+     * @param request incoming request
+     * @return the problem response
+     */
+    default Mono<ResponseEntity<Problem>> create(final Throwable throwable, final ServerWebExchange request) {
```

Comment:
Is it really needed to have Mono as a return type? The docs have examples that are very similar to mvc.

Reply:
For our default implementation, it wouldn't change anything.
But that would give the end-user less power for customization.
If you don't wrap into a Mono, you can't do operations that are blocking (eg. reading from I/O, ...).
Also wrapping into Mono allows to get the [Reactive Context](http://projectreactor.io/docs/core/release/reference/#context) from which you can extract useful things such as the SecurityContext (see [ReactiveSecurityContextHolder](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/core/context/ReactiveSecurityContextHolder.html)) or tracing info, etc...